frontend/src/components/contactsList.tsx

```diff
@@ -0,0 +1,90 @@
+import { useEffect, useState } from 'react'
+import { CreateAContactForm } from './contactsForm'
+import axios from 'axios'
+
+interface Contact {
+  id: string
+  firstName: string
+  lastName?: string
+  email?: string
+  phoneNumber?: string
+  notes?: string
+  source?: string
+  ownerID: string
+  relationships?: UserContactRelationship[]
+  importantDateEvent?: ImportantDateEvent[]
+}
+
+interface UserContactRelationship {
+  id: string
+  relationshipType: string
+}
+
+interface ImportantDateEvent {
+  id: string
+  date: string
+  eventType: string
+}
+
+export const ContactsList = () => {
+  const [showCreateAContactForm, setShowCreateAContactForm] = useState<boolean>(false)
+  const [contacts, setContacts] = useState<Contact[]>([])
+
+  const toggleContactForm = () => {
+    setShowCreateAContactForm(!showCreateAContactForm)
+  }
+
+  const handleContactCreate = (newContact: Contact) => {
+    console.log(newContact)
+    setContacts((prevState) => [...prevState, newContact])
+  }
+
+  useEffect(() => {
+    const getContacts = async () => {
+      try {
+        const response = await axios.get('http://localhost:8000/contacts')
+        const contactsData = response.data.contacts as Contact[]
+        setContacts(contactsData)
+      } catch (error) {
+        console.error(error)
+      }
+    }
+    void getContacts()
```

Comment:
is this intentional? i've never seen this syntax before haha

Reply:
I did this based off the error message when I have getContacts() written on line 52.
Error:
Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator.

Reply:
ah i misread this earlier - i don't know if the catch in the getUserData gets thrown correctly if this line 51 here doesn't also include a .catch block

Reply:
Is this what you mean by including another catch block. Sorry if I'm not understanding this correctly.
 useEffect(() => {
    const getContacts = async () => {
      try {
        const response = await axios.get('http://localhost:8000/contacts', { withCredentials: true })
        const contactsData = response.data.contacts as Contact[]
        setContacts(contactsData)
      } catch (error) {
        console.error(error)
      }
    }
    try {
      void getContacts()
    } catch (error) {
      console.error(error)
    }
  }, [])

Reply:
this gives me the err: Expected an assignment or function call and instead saw an expression.

But this ends up working: getContacts().catch(console.error)

lmk if that works.